weave/trace/vals.py

```diff
@@ -431,7 +487,11 @@ def _remote_iter(self) -> Generator[dict, None, None]:
             page_index += 1
 
     def __getitem__(self, key: Union[int, slice, str]) -> Any:
-        rows = self.rows
+        # TODO: ideally we would have some sort of intelligent
```

Comment:
Agreed.  It's not terrible to put an LRU cache on `self._inefficiently_materialize_rows_as_list`